inner/src/lib.rs

```diff
@@ -131,10 +159,25 @@ mod store {
         };
         // This is the unbounded default.
         match options.lru_max_entries {
-            None => (
-                quote::quote! { std::collections::HashMap<#key_type, #value_type> },
-                quote::quote! { std::collections::HashMap::new() },
-            ),
+            None => {
+				if let Some(hasher) = &options.custom_hasher {
+                    if let Some(hasher_init) = &options.custom_hasher_initializer {
+						return (
+							quote::quote! { #hasher<#key_type, #value_type> },
+							quote::quote! { #hasher_init },
+						);
+					} else {
+						return (
+							quote::quote! { #hasher<#key_type, #value_type> },
+							quote::quote! { #hasher::new() },
+						)
+					}
+                }
+                (
+                    quote::quote! { std::collections::HashMap<#key_type, #value_type> },
+                    quote::quote! { std::collections::HashMap::new() },
+                )
+            }
             Some(cap) => (
```

Comment:
One issue I was worried about with the custom hashers (because before we didn't care about it): it's probably a good idea to raise an error if both an LRU capacity and a custom hasher are given, as we cannot fulfill both. This avoid surprise about the right hasher not being used.

Reply:
This would need an error handling system. I can do it for the next PR and set a simple `panic!(...)` For this one.

Reply:
Yes, that's a good idea. Although the panic is good enough for now as it prevents using it wrong, and still surfaces the reason.